src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -213,7 +213,7 @@ public static final class Facts {
      * field is a DTO for transmission to the client carrying PEM in plain text, and is
      * not stored in the database.
      */
-    @OneToOne
+    @OneToOne(cascade = CascadeType.REMOVE)
```

Comment:
Can we add a unit/integration test (probably in either ConsumerTest or ConsumerCuratorTest ) to test this?

Reply:
Done.